satellite/console/consoleweb/server.go

```diff
@@ -394,17 +395,47 @@ func (s *Server) Run(ctx context.Context) (err error) {
 	var group errgroup.Group
 	group.Go(func() error {
 		<-ctx.Done()
-		return s.server.Shutdown(nil)
+		return server.server.Shutdown(nil)
 	})
 	group.Go(func() error {
 		defer cancel()
-		return s.server.Serve(s.listener)
+		return server.server.Serve(server.listener)
 	})
 
 	return group.Wait()
 }
 
 // Close closes server and underlying listener
-func (s *Server) Close() error {
-	return s.server.Close()
+func (server *Server) Close() error {
+	return server.server.Close()
+}
+
+// gzipHandler is used to gzip static content to minify resources if browser support such decoding
+func (server *Server) gzipHandler(fn http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		isGzipSupported := strings.Contains(r.Header.Get("Accept-Encoding"), "gzip")
+		extension := filepath.Ext(r.RequestURI)
+		// we have gzipped only fonts, js and css bundles
+		formats := map[string]bool{
+			".js":  true,
+			".ttf": true,
+			".css": true,
+		}
+		isNeededFormatToGzip := formats[extension]
+
+		// in case if old browser doesn't support gzip decoding or if file extension is not recommended to gzip
+		// just return original file
+		if !isGzipSupported || !isNeededFormatToGzip {
+			fn.ServeHTTP(w, r)
+			return
+		}
+
+		w.Header().Set("Content-Type", mime.TypeByExtension(extension))
+		w.Header().Set("Content-Encoding", "gzip")
+
+		r.RequestURI += ".gz"
```

Comment:
Requests shouldn't be modified. It can break things. https://golang.org/pkg/net/http/#Handler